### PR TITLE
Use a simplified codepath if no bidi isolation controls are present.

### PR DESCRIPTION
**Pull request by jfkthame**
If there aren't any bidi isolation control characters in the paragraph, each level run directly corresponds to one isolating run sequence. (See notes under https://www.unicode.org/reports/tr9/#BD13.) In this case we can take a simplified codepath to process them.

Also, we can avoid the need for a separate pass over the levels array to find level runs by collecting the level runs at the same time as the levels array is being set up, during explicit::compute().

### CODE REVIEW

File: src/prepare.rs [hunk 1]
```diff
@@ -97,7 +101,7 @@ pub fn isolating_run_sequences(
             };
 
             isolating_run_sequences.push(IsolatingRunSequence {
-                runs: vec![run],
+                runs: vec![run.clone()],
```
Comment (Manishearth) [inline src/prepare.rs:104]: we can use `.drain()` or `into_iter` so we can move instead

File: src/explicit.rs [hunk 2]
```diff
@@ -182,6 +190,22 @@ pub fn compute<'a, T: TextSource<'a> + ?Sized>(
             levels[i + j] = levels[i];
             processing_classes[i + j] = processing_classes[i];
         }
+
+        // Check if we need to start a new level run.
+        if i == 0 {
```
Comment (Manishearth) [inline src/explicit.rs:195]: if we're moving `level_runs` logic here then we should:

 - link to the spec
 - mention in the fn docs that this code also handles that part of the spec
 - ideally, more comments referencing spec list items or other text

File: src/prepare.rs (hunk 1 above)
Comment (jfkthame) [reply src/prepare.rs]: Actually, we can just remove the `.clone()`, it's not needed here.

File: src/explicit.rs (hunk 2 above)
Comment (jfkthame) [reply src/explicit.rs:196]: I've added/tidied up comments, as well as added a test for the edge-case of calling the paragraph API with empty text (and RTL direction, which prevents us taking the early exit in `compute_bidi_info_for_para`).

**Review by Manishearth — APPROVED** [review]
